src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/person/ReclusterService.kt

```diff
@@ -85,14 +126,37 @@ class ReclusterService(
     return matchService.findHighConfidenceMatches(recordsToMatch, PersonSearchCriteria.from(recordToMatch)).map { it.candidateRecord }
   }
 
-  fun <T> addAllIfNotPresent(list: MutableList<T>, elements: List<T>) {
+  private fun mergeRecordToUUID(sourcePersonEntity: PersonEntity, targetPersonEntity: PersonEntity) {
+    val sourcePersonKey = sourcePersonEntity.personKey
+    val targetPersonKey = targetPersonEntity.personKey
+
+    // Step 1: Mark the source key as merged
+    sourcePersonKey?.mergedTo = targetPersonKey?.id
+    sourcePersonKey?.status = UUIDStatusType.RECLUSTER_MERGE
+
+    // Step 2: Detach source entity from its original personKey
+    sourcePersonKey?.personEntities?.remove(sourcePersonEntity)
+    personKeyRepository.save(sourcePersonKey!!)
```

Comment:
if we are going to !! might as well do it on line 130 and save a few ? :-)

Reply:
Good point, Ill do subsequent PR 👍